### PR TITLE
Add get_by_id and get_subservice_by_id

### PR DESCRIPTION
**Pull request by djberg96**
This adds two methods to the ResourceGroupBasedService that will allow users to get objects based on an ID string without having to parse that information out themselves.

For example, a VM contains an ID for its NIC like this:

    "id": "/subscriptions/xyz/resourceGroups/some_group/providers/Microsoft.Network/networkInterfaces/some_nic"

With that information we can then get a model object like so:

    nic_id = vm.properties.network_profile.network_interfaces[0].id
    vms.get_by_id(nic_id)

This will also make deleting associated resources easier.

### CODE REVIEW

**Comment by djberg96** [comment]
@bzwei @bronaghs Please review.

File: lib/azure/armrest/resource_group_based_service.rb [hunk 1]
```diff
@@ -73,6 +73,85 @@ def list_all(filter = {})
...
+      #
+      # Example:
+      #
+      #   vms = Azure::Armrest::VirtualMachineService.new(conf)
+      #
+      #   vm = vms.get('your_vm', 'your_group')
+      #   nic_id = vm.properties.network_profile.network_interfaces[0].id
+      #   nic = vm.get_by_id(nic_id)
+      #
+      def get_by_id(id_string)
```
Comment (bzwei) [inline lib/azure/armrest/resource_group_based_service.rb]: suggest method name `get_associate_resource`

File: lib/azure/armrest/resource_group_based_service.rb [hunk 2]
```diff
@@ -73,6 +73,85 @@ def list_all(filter = {})
...
+        info = parse_id_string(id_string)
+
+        api_version = configuration.provider_default_api_version(info[:provider], info[:service_name])
+        api_version ||= configuration.api_version
+
+        url = File.join(Azure::Armrest::RESOURCE, id_string) + "?api-version=#{api_version}"
+
+        model_class = case info[:service_name].downcase
```
Comment (bzwei) [inline lib/azure/armrest/resource_group_based_service.rb]: let's follow the indent style required by rubocop

Comment (djberg96) [inline lib/azure/armrest/resource_group_based_service.rb]: I think the indent style that rubocop wants is terrible.

File: lib/azure/armrest/resource_group_based_service.rb (hunk 1 above)
Comment (djberg96) [inline lib/azure/armrest/resource_group_based_service.rb]: Hm, ok, but I've kept the originals as aliases because I like short method names. :)

**Comment by djberg96** [comment]
@bzwei Ok, I modified the code so that it's a single method that works with either a service or subservice.

File: lib/azure/armrest/resource_group_based_service.rb [hunk 3]
```diff
@@ -187,6 +238,37 @@ def list_in_all_groups
 
         array
       end
+
+      def get_model_class(service_name)
+        case service_name.downcase
```
Comment (bzwei) [inline lib/azure/armrest/resource_group_based_service.rb]: We can simply maintain a hash table for the lookup instead of long list of case/when

Comment (djberg96) [inline lib/azure/armrest/resource_group_based_service.rb]: @bzwei ok, done.

**Comment by bzwei** [comment]
@djberg96 looks good. Can you address the warnings from rubocop?

**Comment by djberg96** [comment]
@bzwei ok, fixed the rubocop warnings that I could.

**Comment by bronaghs** [comment]
👍

**Pull request merged**